### PR TITLE
Add checkMappings task

### PR DESCRIPTION
**Pull request by modmuss50**
Checks for the issues fixed in https://github.com/FabricMC/yarn/pull/3873

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -150,7 +151,7 @@ tasks.register('yarnCommon', EnigmaTask) {
 	mappings = mappingsDir
 }
 
-tasks.register('checkMappings') {
+def checkPackages = tasks.register('checkPackages') {
```
Comment (liach) [inline build.gradle:158]: Do we need to declare a variable here? I thought we can always just use the task name to refer to the task, or `task.getByName('checkPackages')` in the worst case

File: filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+		for (MappingTree.ClassMapping classMapping : tree.getClasses()) {
+			method: for (MappingTree.MethodMapping method : classMapping.getMethods()) {
+				Set<String> names = new HashSet<>();
+
+				for (int id = minNamespaceId; id < maxNamespaceId; id++) {
+					String name = method.getName(id);
+
+					if (name == null) {
+						break method;
```
Comment (liach) [inline filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java]: I think we should break the `id` loop or continue the method loop instead.

**Comment by Shnupbups** [comment]
Would be nice if this detected duplicate class and field mappings in addition to methods, especially given the recent bug causing renamed classes to not have their old .mapping files deleted.

File: filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java (hunk 2 above)
Comment (NebelNidas) [reply filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java]: Fixed with my refactor 👍

**Review by NebelNidas — APPROVED** [review]

**Review by Shnupbups — APPROVED** [review]

**Comment by Shnupbups** [comment]
Needs developer review (i.e. modmuss or Player) for merge

**Review by Player3324 — APPROVED** [review]

**Comment by Shnupbups** [comment]
Requires #3948 before checks will pass

File: filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.TaskAction;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import net.fabricmc.filament.task.base.FilamentTask;
+import net.fabricmc.mappingio.MappingReader;
```
Comment (liach) [inline filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java:19]: Checkstyle is failing on these out-of-order imports.

Comment (Shnupbups) [reply filament/src/main/java/net/fabricmc/filament/task/mappingio/CheckMappingsTask.java:19]: Fixed

**Comment by Shnupbups** [comment]
Needs dev review again...

**Pull request merged**